httpserver/onion_processing.cpp

```diff
@@ -349,4 +278,74 @@ void RequestHandler::process_onion_req(const std::string& ciphertext,
     }
 }
 
+/// We are expecting a payload of the following shape:
+/// | <4 bytes>: N | <N bytes>: ciphertext | <rest>: json as utf8 |
+auto parse_combined_payload(const std::string& payload) -> CiphertextPlusJson {
+
+    OXEN_LOG(trace, "Parsing payload of length: {}", payload.size());
+
+    auto it = payload.begin();
+
+    /// First 4 bytes as number
+    if (payload.size() < 4) {
+        OXEN_LOG(warn, "Unexpected payload size");
+        throw std::exception();
+    }
+
+    const auto b1 = reinterpret_cast<const uint32_t&>(*it);
+    const auto n = boost::endian::little_to_native(b1);
```

Comment:
This reinterpret_cast is undefined behaviour (though it works on x86) without an alignment guarantee, and the copy via boost is unnecessary since you never care about b1 (and boost has a nicer no-op-on-little-endian conversion function):

```suggestion
    uint32_t n;
    std::memcpy(&n, payload.data(), sizeof(uint32_t));
    boost::endian::little_to_native_inplace(n);
```

Reply:
updated